### PR TITLE
Handle transient unmount errors at test teardown

### PR DESCRIPTION
**Pull request by jmmv**
It is possible, although rare, for a system component such as macOS's
Finder to interfere with the sandboxfs test instances we set up.  One
of the interferences we observe is that the Finder may access the
file system while we are running the test teardown routine, and thus
the unmount we issue fails due to a "resource busy" error.

Under those conditions, the only thing we can do is retry the unmount
operation because the fact that it failed is not our fault.

This should fix some very rare test flakiness I've observed on Travis
and also resolves some consistent flakiness I was observing in new
tests I'm adding, which left me puzzled for a long time.

### CODE REVIEW

File: integration/utils/exec.go [hunk 1]
```diff
@@ -410,12 +414,24 @@ func mountSetupFull(t *testing.T, stdout io.Writer, stderr io.Writer, user *Unix
 // If tests wish to control the shutdown of the sandboxfs process, they can do so, but then they
 // must set s.Cmd to nil to tell TearDown to not clean up the process a second time.  The same
 // applies to s.Stdin.
-func (s *MountState) TearDown(t *testing.T) {
+//
+// If tests wish to check if TearDown returned an error, they can do so by avoiding the recommended
+// use of "defer".  Note, though, that such tests will only receive the first error encountered by
+// this function, and that the function will run to completion even if there were failures.
+func (s *MountState) TearDown(t *testing.T) error {
+	var finalErr error
```
Comment (rupertks) [inline integration/utils/exec.go]: Call this firstErr instead? 

That seems to be what it is, and the comment above talks about "first error" too

Comment (jmmv) [reply integration/utils/exec.go]: Done. "final" here meant "error to be returned in the end".

**Comment by philwo** [comment]
> Under those conditions, the only thing we can do is retry the unmount
operation because the fact that it failed is not our fault.

What would happen if we accidentally left a process running in that directory? For example, if a test spawns a daemon that never exits. (process-wrapper cannot prevent this at the moment, there is no known mechanism on macOS to track daemonizing child processes)

**Comment by jmmv** [comment]
Note that this fix is for our own tests, not a fix in sandboxfs itself.

If we leak a process from our own tests, then the various retries to unmount during teardown will fail and the test will abort. I think this is acceptable given that this is a bug on our side.

If this happens from Bazel where a process is left behind as you describe... well, the mount point will be left busy and Bazel won't be able to unmount it. That will need a different fix (to code that's not yet submitted!), and we can probably resolve that from Bazel's side. Note that BSDs (and thus macOS) have an "force" flag to the unmount system call / command, which causes any active file descriptors to be revoked. This should work for the case you are describing.

**Comment by philwo** [comment]
Oh, sorry! Of course. In that case this is fine :)

**Pull request merged**